react-app/src/components/RHF/FieldArray.tsx

```diff
@@ -18,15 +18,23 @@ export const RHFFieldArray = <TFields extends FieldValues>(props: FieldArrayProp
     fieldArr.append(props.fields.reduce(slotInitializer(), {}) as never);
   };
 
+  // on-load or if the element id changes, scroll this element into view
   useEffect(() => {
-    if (fieldArr.fields.length) return;
-    fieldArr.append(props.fields.reduce(slotInitializer(), {}) as never, {
-      shouldFocus: false,
-    });
-  }, [fieldArr, props.fields]);
+    const element = document.getElementById(props.name);
+    if (element) {
+      element.scrollIntoView({ behavior: "smooth", block: "center", inline: "nearest" });
+    }
+  }, [props.name]);
+
+  // if there are no fields in the field array set it to the props.field
+  useEffect(() => {
+    if (fieldArr.fields.length == 0) {
+      fieldArr.append(props.fields.reduce(slotInitializer(), {}) as never);
+    }
+  }, [props.fields]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Can you add a comment of what's wrong?

Reply:
if you are triggering it on change, it triggers when initially and if the fieldArr doesn't have values, it adds them and then that triggers a change and sometimes gets added again depending on timing

Reply:
I meant in the actual code...

Reply:
I must admit that I don't know what the underlying issue was. I looked pretty hard, but I wasn't able to find something. I assume it is HTML weirdness. I can keep working to track it down, but I figured we could push this fix in the meantime